src/conditional-compilation.md

```diff
@@ -22,43 +22,31 @@
 > _ConfigurationPredicateList_\
 > &nbsp;&nbsp; _ConfigurationPredicate_ (`,` _ConfigurationPredicate_)<sup>\*</sup> `,`<sup>?</sup>
 
-*Conditionally compiled source code* is source code that may or may not be
-considered a part of the source code depending on certain conditions. <!-- This
-definition is sort of vacuous --> Source code can be conditionally compiled
-using the [attributes] [`cfg`] and [`cfg_attr`] and the built-in [`cfg` macro].
-These conditions are based on the target architecture of the compiled crate,
-arbitrary values passed to the compiler, and a few other miscellaneous things
-further described below in detail.
+*Conditionally compiled source code* is source code that is compiled only under certain conditions. Source code can be made conditionally compiled using the [`cfg`] and [`cfg_attr`] [attributes] and the built-in [`cfg` macro]. Whether to compile can depend on the target architecture of the compiled crate, arbitrary values passed to the compiler, and other things further described below.
 
 Each form of conditional compilation takes a _configuration predicate_ that
 evaluates to true or false. The predicate is one of the following:
 
-* A configuration option. It is true if the option is set and false if it is
+* A configuration option. The predicate is true if the option is set, and false if it is
   unset.
-* `all()` with a comma separated list of configuration predicates. It is false
-  if at least one predicate is false. If there are no predicates, it is true.
-* `any()` with a comma separated list of configuration predicates. It is true
-  if at least one predicate is true. If there are no predicates, it is false.
-* `not()` with a configuration predicate. It is true if its predicate is false
-  and false if its predicate is true.
+* `all()` with a comma-separated list of configuration predicates. It is true precisely if all of the given predicates are true.
+* `any()` with a comma-separated list of configuration predicates. It is true precisely if at least one of the given predicates is true.
+* `not()` with a configuration predicate. It is true precisely if the given predicate is false.
```

Comment:
I don't feel like brevity of words here is helping. The original seemed sufficiently brief, and seemed a little clearer to me. I would suggest keeping this as it was before.

```suggestion
* * `not()` with a configuration predicate. It is true if its predicate is false
  and false if its predicate is true.
```